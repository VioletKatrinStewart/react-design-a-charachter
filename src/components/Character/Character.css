.div1 {
  background-color: yellow;
}

.head {
  background-color: yellow;
}
