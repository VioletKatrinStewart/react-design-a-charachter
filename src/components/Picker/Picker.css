.characterImg {
  width: 100px;
}

.imgs {
  background-color: palegoldenrod;
  display: flex;
  flex-direction: column;
  width: 500px;
  align-items: center;
  margin: auto;
}
