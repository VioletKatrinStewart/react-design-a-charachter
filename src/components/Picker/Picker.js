import React from 'react';
export default function Picker({ head, middle, pants }) {
  return (
    <>
      <div>
        <img src={`${process.env.PUBLIC_URL}/head/bird-head.png`} />
        <img src={`${process.env.PUBLIC_URL}/head/${head}.png`} />
      </div>
      <div>
        <img src={`${process.env.PUBLIC_URL}/middle/blue-middle.png`} />
        <img src={`${process.env.PUBLIC_URL}/middle/${middle}.png`} />
      </div>
      <div>
        <img src={`${process.env.PUBLIC_URL}/leg-pants.png`} />
        <img src={`${process.env.PUBLIC_URL}/pants/${pants}.png`} />
      </div>
    </>
  );
}
