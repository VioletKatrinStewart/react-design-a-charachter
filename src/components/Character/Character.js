import React from 'react';
import './Character.css';
export default function CharacterOptions({
  head,
  setHead,
  setHeadCount,
  middle,
  setMiddle,
  setMiddleCount,
  pants,
  setPants,
  setPantsCount,
  catchPhrase,
  setCatchPhrase,
  setCatchPhraseList,
}) {
  const handleHead = (e) => {
    setHead(e.target.value);
    setHeadCount((prevState) => prevState + 1);
  };
  const handleMiddle = (e) => {
    setMiddle(e.target.value);
    setMiddleCount((prevState) => prevState + 1);
  };
  const handlePants = (e) => {
    setPants(e.target.value);
    setPantsCount((prevState) => prevState + 1);
  };
  const handleCatchPhrase = () => {
    setCatchPhraseList((prevState) => [...prevState, catchPhrase]);
    setCatchPhrase('');
  };
  return (
    <div>
      <div className="div1">
        <label className="head">
          Head
          <select value={head} onChange={handleHead}>
            <option value="dog-head">Dog</option>
            <option value="bird-head">Bird</option>
            <option value="duck-head">Duck</option>
            <option value="horse-head">Horse</option>
          </select>
        </label>
        <label>
          Shirt
          <select value={middle} onChange={handleMiddle}>
            <option value="red-middle">Red Shirt</option>
            <option value="pink-middle">Pink Shirt</option>
            <option value="blue-middle">Blue Shirt</option>
            <option value="dress-middle">Dress</option>
          </select>
        </label>
        <label>
          Pants
          <select value={pants} onChange={handlePants}>
            <option value="leg-pants">Leg</option>
            <option value="blue-pants">Blue Pants </option>
            <option value="dog-pants">Dog Pants</option>
            <option value="white-pants">White Pants</option>
          </select>
        </label>
        <label>
          What`s your catch phrase?
          <input
            className="catchphrase"
            type="text"
            value={catchPhrase}
            onChange={(e) => setCatchPhrase(e.target.value)}
          />
          <button onClick={handleCatchPhrase}>Add</button>
        </label>
      </div>
    </div>
  );
}
