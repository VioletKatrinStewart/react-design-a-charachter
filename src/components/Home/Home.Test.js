import { render } from '@testing-library/react';
import Picker from '../Picker/Picker';
import Character from '../Character/Character';

test('renders picker component', () => {
  const container = render(
    <div>
      <Picker head={'bird-head'} middle={'pink-middle'} pants={'blue-pants'} />
      <Character head="duck" middle="pink" bottom="blue" />
    </div>
  );
  expect(container).toMatchSnapshot();
});
// hello
