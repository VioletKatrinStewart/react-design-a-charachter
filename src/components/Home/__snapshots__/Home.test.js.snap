// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders picker component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="imgs"
        >
          <img
            class="characterImg"
            src="/head/bird-head.png"
          />
          <img
            class="characterImg"
            src="/middle/pink-middle.png"
          />
          <img
            class="characterImg"
            src="/pants/blue-pants.png"
          />
        </div>
        <div>
          <div
            class="div1"
          >
            <label
              class="head"
            >
              Head
              <select>
                <option
                  value="dog-head"
                >
                  Dog
                </option>
                <option
                  value="bird-head"
                >
                  Bird
                </option>
                <option
                  value="duck-head"
                >
                  Duck
                </option>
                <option
                  value="horse-head"
                >
                  Horse
                </option>
              </select>
            </label>
            <label>
              Shirt
              <select>
                <option
                  value="red-middle"
                >
                  Red Shirt
                </option>
                <option
                  value="pink-middle"
                >
                  Pink Shirt
                </option>
                <option
                  value="blue-middle"
                >
                  Blue Shirt
                </option>
                <option
                  value="dress-middle"
                >
                  Dress
                </option>
              </select>
            </label>
            <label>
              Pants
              <select>
                <option
                  value="leg-pants"
                >
                  Leg
                </option>
                <option
                  value="blue-pants"
                >
                  Blue Pants 
                </option>
                <option
                  value="dog-pants"
                >
                  Dog Pants
                </option>
                <option
                  value="white-pants"
                >
                  White Pants
                </option>
              </select>
            </label>
            <label>
              What\`s your catch phrase?
              <input
                class="catchphrase"
                type="text"
                value=""
              />
              <button>
                Add
              </button>
            </label>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="imgs"
      >
        <img
          class="characterImg"
          src="/head/bird-head.png"
        />
        <img
          class="characterImg"
          src="/middle/pink-middle.png"
        />
        <img
          class="characterImg"
          src="/pants/blue-pants.png"
        />
      </div>
      <div>
        <div
          class="div1"
        >
          <label
            class="head"
          >
            Head
            <select>
              <option
                value="dog-head"
              >
                Dog
              </option>
              <option
                value="bird-head"
              >
                Bird
              </option>
              <option
                value="duck-head"
              >
                Duck
              </option>
              <option
                value="horse-head"
              >
                Horse
              </option>
            </select>
          </label>
          <label>
            Shirt
            <select>
              <option
                value="red-middle"
              >
                Red Shirt
              </option>
              <option
                value="pink-middle"
              >
                Pink Shirt
              </option>
              <option
                value="blue-middle"
              >
                Blue Shirt
              </option>
              <option
                value="dress-middle"
              >
                Dress
              </option>
            </select>
          </label>
          <label>
            Pants
            <select>
              <option
                value="leg-pants"
              >
                Leg
              </option>
              <option
                value="blue-pants"
              >
                Blue Pants 
              </option>
              <option
                value="dog-pants"
              >
                Dog Pants
              </option>
              <option
                value="white-pants"
              >
                White Pants
              </option>
            </select>
          </label>
          <label>
            What\`s your catch phrase?
            <input
              class="catchphrase"
              type="text"
              value=""
            />
            <button>
              Add
            </button>
          </label>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
